pkg/certificateprovider/certificate_provider.go

```diff
@@ -12,5 +13,30 @@ type CertStoreConfig map[string]string
 // CertificateProvider is an interface that defines methods to be implemented by a each certificate provider
 type CertificateProvider interface {
 	// Returns an array of certificates based on certificate properties defined in attrib map
-	GetCertificatesContent(ctx context.Context, attrib map[string]string) ([]types.Certificate, error)
+	GetCertificates(ctx context.Context, attrib map[string]string) ([]*x509.Certificate, error)
+}
+
+// Decode PEM-encoded bytes into an x509.Certificate chain.
+func DecodeCertificates(value []byte) ([]*x509.Certificate, error) {
+	var certs []*x509.Certificate
+	block, rest := pem.Decode(value)
+	if block == nil && len(rest) > 0 {
+		return nil, fmt.Errorf("failed to decode pem block")
+	}
+
+	for block != nil {
+		if block.Type == "CERTIFICATE" {
```

Comment:
what if the type is not `CERTIFICATE`? Would it fall into an infinite loop?

Reply:
Thanks for this! I wanted to make sure to ignore any private keys, even if they were given to us. My closing brace is clearly in the wrong place 😅

Fixing + adding a test